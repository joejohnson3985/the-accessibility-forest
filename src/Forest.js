import React, { Component } from 'react';
import './Forest.scss';

class Forest extends Component {
  constructor(props) {
    super(props);
    this.state ={
    }
  }

  render() {
    let whatToRender;
    if(this.props.forestName) {
      whatToRender = this.props.forestName
    } else {
      whatToRender = 'Your nameless forest'
    }

    return (
      <div>
        <section className='forest'>
          <table className="trees">
            <caption><strong>{whatToRender}</strong></caption>
            <tbody>
              <tr>
                {this.props.trees}
              </tr>
            </tbody>
          </table>
        </section>
      </div>
    )
  }
}

export default Forest;