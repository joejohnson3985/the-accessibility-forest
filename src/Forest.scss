@import './_variables.scss';

.forest {
  width: 100%;
  background-color: $plain-background;
  @include center-flex;
  height: 20vh;
}
.trees{
  width: 500px;
}

.trees td {
  text-align: center;
  height: 100px;
  width: 100px;
  margin: 4px;
}

.trees td img {
  // visibility: hidden;
  height: 100px;
}

.trees caption {
  margin-top: 10px;
  font-size: 16px;
  color: $primary-color;
}