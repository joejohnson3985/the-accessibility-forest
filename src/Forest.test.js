import React from 'react';
import Forest from './Forest';
import { shallow } from 'enzyme';
import Tree1 from './Images/tree1.png'


const mockTrees = [<td key={'test-tree'}><img alt='Geometric tree illustration' src={Tree1}/></td>]
const mockPoints = 2
const mockName = "Joe's forest"

describe('Forest', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
        <Forest
          trees={mockTrees} 
          points={mockPoints}
          forestName={mockName}
        />
      )
  });

  it('Should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})

describe('Forest no name', () => {
  const noMockName= ""
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
        <Forest
          trees={mockTrees} 
          points={mockPoints}
          forestName={noMockName}
        />
      )
  });

  it('Should match the snapshot without name', () => {
    expect(wrapper).toMatchSnapshot();
  })
})