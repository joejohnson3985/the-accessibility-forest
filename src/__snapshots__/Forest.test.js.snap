// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forest Should match the snapshot 1`] = `
<div>
  <section
    className="forest"
  >
    <table
      className="trees"
    >
      <caption>
        <strong>
          Joe's forest
        </strong>
      </caption>
      <tbody>
        <tr>
          <td
            key="test-tree"
          >
            <img
              alt="Geometric tree illustration"
              src="tree1.png"
            />
          </td>
        </tr>
      </tbody>
    </table>
  </section>
</div>
`;

exports[`Forest no name Should match the snapshot without name 1`] = `
<div>
  <section
    className="forest"
  >
    <table
      className="trees"
    >
      <caption>
        <strong>
          Your nameless forest
        </strong>
      </caption>
      <tbody>
        <tr>
          <td
            key="test-tree"
          >
            <img
              alt="Geometric tree illustration"
              src="tree1.png"
            />
          </td>
        </tr>
      </tbody>
    </table>
  </section>
</div>
`;
